bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/link/ThingLinkManager.java

```diff
@@ -46,17 +46,19 @@
     private boolean autoLinks = true;
 
     protected void activate(ComponentContext context) {
+        modified(context);
+        itemChannelLinkRegistry.addRegistryChangeListener(itemChannelLinkRegistryChangeListener);
+        managedThingProvider.addProviderChangeListener(managedThingProviderListener);
+    }
 
+    protected void modified(ComponentContext context) {
         // check whether we want to enable the automatic link creation or not
         if (context != null) {
             Object value = context.getProperties().get("autoLinks");
-            if ("false".equals(value)) {
+            if (value != null && value.toString().equals("false")) {
```

Comment:
If the configuration has been set to false and this configuration is removed again, the default value should be used.
The current implementation uses the default / true on restart but not if it gets modified.

If I am correct you should perhaps use:
<code>autoLinks = value != null && value.toString().equals("false");</code>